### PR TITLE
[BX-1294] Fix missing wallets in Send flow

### PR DESCRIPTION
**Pull request by BrodyHughes**
Fixes BX-1294

## What changed (plus any additional context for devs)
- was previously just checking the order of wallets in `useWalletOrderStore()` and returning those. this was only ever updated when the order was changed in the walletSwitcher.
- I updated the send flow to check the existing list of wallets in `getAccounts()` and then compare that to the order of the wallets in `useWalletOrderStore()`
- it then updates the `useWalletOrderStore()` with all the wallets in the correct order and also updated the `useWalletOrderStore()`

## What to test
- see if send suggestions includes all the wallets in your bx aside from the current wallet
- make sure code makes sense

## Screen recordings / screenshots

<video src="https://github.com/rainbow-me/browser-extension/assets/41711440/1d09df63-322f-4255-8be3-dbe94c520328" >

### CODE REVIEW

File: src/entries/popup/pages/send/ToAddressInput.tsx [hunk 1]
```diff
@@ -161,15 +162,45 @@ const DropdownWalletsList = ({
...
   selectWalletAndCloseDropdown: (address: Address) => void;
 }) => {
-  const { walletOrder } = useWalletOrderStore();
+  const [newWalletOrder, setNewWalletOrder] = useState<`0x${string}`[]>([]);
+  const { walletOrder, saveWalletOrder } = useWalletOrderStore();
+
+  // find the items in getAccounts then compare to the current
+  // walletOrder, then return all items in getAccounts in order
+
+  useEffect(() => {
```
Comment (greg-schrammel) [inline src/entries/popup/pages/send/ToAddressInput.tsx:171]: have you tried the `useAccounts` hook?

**Review by DanielSinclair — APPROVED** [review]
Functionality looks good to me; didn't review code

File: src/entries/popup/pages/send/ToAddressInput.tsx [hunk 2]
```diff
@@ -161,15 +162,45 @@ const DropdownWalletsList = ({
...
-    [wallets, walletOrder],
+    () => sortWallets(newWalletOrder, wallets),
+    [newWalletOrder, wallets],
   );
   const sortedWatchedWallets = useMemo(
-    () => sortWallets(walletOrder, watchedWallets),
-    [watchedWallets, walletOrder],
+    () => sortWallets(newWalletOrder, watchedWallets),
+    [newWalletOrder, watchedWallets],
   );
```
Comment (greg-schrammel) [inline src/entries/popup/pages/send/ToAddressInput.tsx:202]: after playing with it a bit I'm pretty sure the props are in the right state already, we just need to remove all sorting from inside this component
like 
Screenshot 2024-03-07 at 15 46 33

Comment (BrodyHughes) [reply src/entries/popup/pages/send/ToAddressInput.tsx]: changed lol

**Review by greg-schrammel — APPROVED** [review]

**Pull request merged**